Freebsd port

adriaandegroot opened the pull request:
This PR tries to add downstream packaging fixes -- in this case, makefile settings -- to upstream so that we can simplify packaging. That makes it slightly more likely that changes like GTK3 will "automatically" end up in FreeBSD packaging. I appreciate that upstream probably doesn't have facilities to actually test these builds, though.

While here, add some minor documentation for FreeBSD builds, as if it is Linux (it is UNIX-like, after all).

PaulBoersma reviewed (commented):
I like improvements, but I also like to understand, in order to be able to maintain. Why is gmake necessary instead of make?

PaulBoersma reviewed (commented):
I understand your definitions of CC and CXX, but why do we need to include most Praat source folders in CXXFLAGS?

adriaandegroot commented:
I'll re-vamp this one: you are right in asking why those specific `Makefile` settings are needed and now I look more closely, I'm not sure why. Re-titling as *[WIP]* (work-in-progress).

adriaandegroot commented:
Can be much-less-complicated; this builds with BSD make, although it needs the *dos2unix* branch to be merged to fully work.

adriaandegroot commented:
Rebased for current *master*. New things are: I set `-Werror=return-type` in order to pick up missing return statements; that's something that I see often(-ish) in code with #ifdefs (separate PR for fixing the one problem that shows up, although in this case it's more that Clang doesn't know that `Melder_fatal()` doesn't return.

The new opus file support was partly-removed(?)  in 94a82a9d58a88865aa1ebea7c31794dceda56c95, but the functions from the library are still called so I'm not sure how this builds on Linux (or anywhere). Added the library to LIBS in order to complete the build on FreeBSD.
